### PR TITLE
Docs(GraphQL)Document the `between` function 

### PR DESCRIPTION
**Pull request by aaroncarey**
Adding documentation for the `between` filter for GraphQL, also minor wording changes to the rest of this page.

Updates to search-filtering.md to document the `between` function in GraphQL:

- Added coverage of the `between` keyword in GraphQL, along with syntax examples from the related code PR (https://github.com/dgraph-io/dgraph/pull/6651)
- Minor rewording throughout

Minor updates to functions.md (DQL) to note that `between` searches in an inclusive range.

@minhaj-shakeel , please let me know if we need to add any caveats or other notes.

Fixes DGRAPH-2607

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6813)

 
 
 
 

Docs Preview: [Dgraph Preview](https://dgraph-2ad448189b-105832.surge.sh)

### CODE REVIEW

File: wiki/content/graphql/queries/search-filtering.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ objects for a type.
 
 ### Get a single object
 
-Fetch the title, text and datePublished for a post with id `0x1`.
+Fetch the title, text and `datePublished` for a post with id `0x1`.
```
Comment (bucanero) [inline wiki/content/graphql/queries/search-filtering.md:14]:
```suggestion
Fetch the `title`, `text`, and `datePublished` for a post with id `0x1`.
```

File: wiki/content/graphql/queries/search-filtering.md [hunk 2]
```diff
@@ -45,7 +44,7 @@ query {
 
 While fetching nested linked objects, you can also apply a filter on them.
 
-Example - Fetching author with id 0x1 and their posts about GraphQL.
+For example, the following query fetches the author with the `id` 0x1 and their posts about GraphQL.
```
Comment (bucanero) [inline wiki/content/graphql/queries/search-filtering.md:47]:
```suggestion
For example, the following query fetches the `author` with the `id` `0x1` and their posts about GraphQL.
```

File: wiki/content/graphql/queries/search-filtering.md [hunk 3]
```diff
@@ -64,11 +63,11 @@ query {
 }
 ```
 
-If your type has a field with `@id` directive on it, you can also fetch objects using that.
+If your type has a field with the `@id` directive applied to it, you can also fetch objects using that.
 
-Example: To fetch a user's name and age by userID which has @id directive.
+For example, given the following schema, the query below fetches a user's name and age by `userID` (which has the `@id` directive):
```
Comment (bucanero) [inline wiki/content/graphql/queries/search-filtering.md:68]:
```suggestion
For example, given the following schema, the query below fetches a user's `name` and `age` by `userID` (which has the `@id` directive):
```

File: wiki/content/graphql/queries/search-filtering.md [hunk 4]
```diff
@@ -89,9 +88,9 @@ query {
 }
 ```
 
-### Query list of objects
+### Query a list of objects
 
-Fetch the title, text and and datePublished for all the posts.
+You can query a list of objects using GraphQL. For example, the following query fetches the title, text and and `datePublished` for all posts:
```
Comment (bucanero) [inline wiki/content/graphql/queries/search-filtering.md:93]:
```suggestion
You can query a list of objects using GraphQL. For example, the following query fetches the `title`, `text`, and `datePublished` for all posts:
```

File: wiki/content/graphql/queries/search-filtering.md [hunk 5]
```diff
@@ -143,10 +140,11 @@ query {
 }
 ```
 
+### Filter a query for a list of objects
+
 You can also filter nested objects while querying for a list of objects.
 
-Example - To fetch all the authors whose name have `Lee` in them and their`completed` posts
-with score greater than 10.
+For example, the following query fetches all of the authors whose name contains `Lee` and with their `completed` posts that have a score greater than 10:
```
Comment (bucanero) [inline wiki/content/graphql/queries/search-filtering.md:147]:
```suggestion
For example, the following query fetches all of the authors whose name contains `Lee` and with their `completed` posts that have a score greater than `10`:
```

File: wiki/content/graphql/queries/search-filtering.md [hunk 6]
```diff
@@ -170,4 +168,44 @@ query {
...
 }
-```
\ No newline at end of file
+```
+
+### Filter a query for a range of objects with `between`
+
+You can also filter query results within an inclusive range of indexed and typed
+scalar values using the `between` keyword. This keyword is also supported for
+DQL; to learn more, see [DQL Functions: `between`](/query-language/functions/#between).
```
Comment (bucanero) [inline wiki/content/graphql/queries/search-filtering.md:177]: This note could be in a green **Tip** box:
```
This keyword is also supported for DQL; to learn more, see [DQL Functions: `between`](/query-language/functions/#between).
```

**Review by bucanero — COMMENTED** [review]
Shared some formatting suggestions.

**Review by aaroncarey — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6813)* status: 0 of 2 files reviewed, 6 unresolved discussions (waiting on @aaroncarey, @bucanero, @danielmai, @MichaelJCompton, and @minhaj-shakeel)

___
*[wiki/content/graphql/queries/search-filtering.md, line 14 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6813#-MKr28gEzAYPhB_LRVtR-r1-14:-MKvdlAM6P1cxXbho1cO:boyexb3) ([raw file](https://github.com/dgraph-io/dgraph/blob/b5951baf214318fa3d775afeaf972b1b9551685d/wiki/content/graphql/queries/search-filtering.md#L14)):*
<details><summary><i>Previously, bucanero (Damián Parrino) wrote…</i></summary><blockquote>

```suggestion
Fetch the `title`, `text`, and `datePublished` for a post with id `0x1`.
```
</blockquote></details>

Done. Thanks!
___
*[wiki/content/graphql/queries/search-filtering.md, line 47 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6813#-MKr28gEzAYPhB_LRVtR-r1-47:-MKve-8mEEjPM20mPGL2:bmni7ff) ([raw file](https://github.com/dgraph-io/dgraph/blob/b5951baf214318fa3d775afeaf972b1b9551685d/wiki/content/graphql/queries/search-filtering.md#L47)):*
<details><summary><i>Previously, bucanero (Damián Parrino) wrote…</i></summary><blockquote>

```suggestion
For example, the following query fetches the `author` with the `id` `0x1` and their posts about `GraphQL`.
```
</blockquote></details>

I applied the `GraphQL` formatting fix, but "author" doesn't appear in the example, so I didn't apply code formatting to it.
___
*[wiki/content/graphql/queries/search-filtering.md, line 68 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6813#-MKr28gEzAYPhB_LRVtR-r1-68:-MKveDk84y3loytSk0ec:b-dtkds3) ([raw file](https://github.com/dgraph-io/dgraph/blob/b5951baf214318fa3d775afeaf972b1b9551685d/wiki/content/graphql/queries/search-filtering.md#L68)):*
<details><summary><i>Previously, bucanero (Damián Parrino) wrote…</i></summary><blockquote>

```suggestion
For example, given the following schema, the query below fetches a user's `name` and `age` by `userID` (which has the `@id` directive):
```
</blockquote></details>

Done, thanks!
___
*[wiki/content/graphql/queries/search-filtering.md, line 93 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6813#-MKr28gEzAYPhB_LRVtR-r1-93:-MKveOW85DWw4iPJOgux:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/b5951baf214318fa3d775afeaf972b1b9551685d/wiki/content/graphql/queries/search-filtering.md#L93)):*
<details><summary><i>Previously, bucanero (Damián Parrino) wrote…</i></summary><blockquote>

```suggestion
You can query a list of objects using GraphQL. For example, the following query fetches the `title`, `text`, and `datePublished` for all posts:
```
</blockquote></details>

Done.
___
*[wiki/content/graphql/queries/search-filtering.md, line 177 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6813#-MKr28gEzAYPhB_LRVtR-r1-177:-MKvevC4-Zh_oKUM_rSv:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/b5951baf214318fa3d775afeaf972b1b9551685d/wiki/content/graphql/queries/search-filtering.md#L177)):*
<details><summary><i>Previously, bucanero (Damián Parrino) wrote…</i></summary><blockquote>

This note could be in a green **Tip** box:
```
This keyword is also supported for DQL; to learn more, see [DQL Functions: `between`](/query-language/functions/#between).
```

</blockquote></details>

Done.

**Review by minhaj-shakeel — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6813)* status: 0 of 2 files reviewed, 7 unresolved discussions (waiting on @aaroncarey, @bucanero, @danielmai, @MichaelJCompton, and @minhaj-shakeel)

___
*[wiki/content/query-language/functions.md, line 405 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/6813#-MKx2DOf6EgpdXJm_oEt:-MKx2DOg7Kl9S37Kummc:b-rrbzht) ([raw file](https://github.com/dgraph-io/dgraph/blob/be60c9c972232a8cf221c0640e2ac8b256ded983/wiki/content/query-language/functions.md#L405)):*
> ```Markdown
> Schema Types: Scalar types, including `dateTime`, `int`, `float` and `string`
> 
> Index Required: `dateTime`
> ```

Could mention indices for other scalar types also Like:
Index Required: `dateTime`, `int`, `float`, and `exact` on strings.
 
and the below example should start as:
"Here is the example usage on `dateTime` ".

**Review by aaroncarey — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6813)* status: 0 of 2 files reviewed, 7 unresolved discussions (waiting on @aaroncarey, @bucanero, @danielmai, @MichaelJCompton, and @minhaj-shakeel)

___
*[wiki/content/query-language/functions.md, line 405 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/6813#-MKx2DOf6EgpdXJm_oEt:-ML9IKKGDwfcRbwtdJkT:b-jsd9r5) ([raw file](https://github.com/dgraph-io/dgraph/blob/be60c9c972232a8cf221c0640e2ac8b256ded983/wiki/content/query-language/functions.md#L405)):*
<details><summary><i>Previously, minhaj-shakeel wrote…</i></summary><blockquote>

Could mention indices for other scalar types also Like:
Index Required: `dateTime`, `int`, `float`, and `exact` on strings.
 
and the below example should start as:
"Here is the example usage on `dateTime` ".
</blockquote></details>

Great feedback, thank you! I updated the "Index Required" section exactly as you suggested, and added a lead-in for the example that clarifies that the example only shows how to use `between` with a `dateTime` index, but that there are other use cases.

**Pull request merged**